### ISSUE TITLE
Feature Request - Allow removal of input files and disabling of tracks with Delete key.

### ISSUE THREAD

**Issue by SterlingMcCall** [mbunkus/mkvtoolnix#1225]
All in the title. I would like to be able to remove files from the files list by selecting them and clicking delete. I would also like to be able to set "Mux this track" to no by selecting tracks and clicking delete. Perhaps even make it toggle between yes and no.

**Comment by mbunkus** [comment]
File removal I'm OK with.

Toggling »mux this« with delete: not so much as it would be very counter-intuitive (usually delete actually removes something). I could get behind using enter and/or a double-click with the mouse. Space would also be a contender, but space is usually used for toggling element selection in such list or tree views, hence enter.

**Comment by remuxer32** [comment]
**»Enable/disable a single track using the Enter key on the keyboard«** is, in my humble opinion, the best way to toggle »mux this«. The Enter key is without a doubt the most popular and useful button on the keyboard. It's located in the centre of the keyboard; that's why people can use it **fast & nicely** ☺. Thank you for considering my hint :).

**Comment by SterlingMcCall** [comment]
I don't have any particularly strong feelings about which key it should be. I just really want to be able to use a keyboard shortcut to do it.

**Comment by remuxer32** [comment]
My strong feelings are located to the Enter key, of course. Believe me: it's the most logical solution to toggle »mux this«. Most people should be satisfied because of that. Usually the Enter key is the largest key on the keyboard. That's why it's the most intuitive and pleasurable button for actions for tracks.

**Comment by bastik-tor** [comment]
Using the delete key for file removal seems to be not controversial.

Using enter (or some other button, though enter seems to be the best) to toggle an option should be consistent. The forced track flag is on or off as well and if enter is limited to on/off settings it should work on that option as well.  The default track flag has three options, so I am unsure if enter would cycle through them or not, but how would a user know if enter has any function bound to it or not.

Currently if an entry is selected pressing space (on Windows) reveals all the options for that entry and it can be selected with the arrow keys. It seems good to keep that the way it is. So not using space for toggling.

**Comment by mbunkus** [comment]
I'm pretty sure that the original request was to have the enter key as a toggle when the focus is on the tracks list. The intention is to be able to toggle those tracks quickly without having to move the mouse to the right. Toggling »mux this« is a valid candidate for this as it's often used whereas other flags like »forced track« aren't.

What you, @bastik-tor , are describing is the behavior if the focus is on the control on the right side. That behavior will not change.

**Comment by bastik-tor** [comment]
>...toggle when the focus is on the tracks list

Sorry I misinterpreted it, maybe because I remembered that you, @mbunkus, did not intend to add anything that is hard to discover on the track list.

>> (me) Since stuff works with a mouse, I double-clicked on the "Mux this"
 check mark with a written yes to try to toggle it to an red cross with
 a written no.

> (@mbunkus) I don't want to have hidden functionality in the tree views because it
is even less easy to discover than the context menus.

That led me to the conclusion that only the action on the right side was under discussion.

Arguably it is the where to double-click or when to press a button, that is different from what I tried "back then." Never mind I am all for having anything to toggle it easily, like a key press and/or a double click.
